webserver/__init__.py

```diff
@@ -20,24 +31,35 @@ def create_app_with_configuration(config_path=None):
     )
 
     # Configuration
-    root_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), "..")
-    app.config.from_pyfile(os.path.join(
-        root_path,
-        'default_config.py'
-    ))
-    app.config.from_pyfile(os.path.join(
-        root_path,
-        'custom_config.py'
-    ), silent=True)
+
+    # if deploying in a production docker environment, load consul_config.py
+    if deploy_env and not config_path:
+        print("Checking if consul generated config file exists: $s" % config_file)
+        consul_config = os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "consul_config.py")
+        for _ in range(CONSUL_CONFIG_FILE_RETRY_COUNT):
+            if not os.path.exists(config_file):
+                time.sleep(1)
+
+        if not os.path.exists(config_file):
+            print("No config file generated. Retried %d times, exiting." % CONSUL_CONFIG_FILE_RETRY_COUNT)
+            sys.exit(-1)
+
+        app.config.from_pyfile(consul_config)
+        return app
+
+    config_file = os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "config.py")
```

Comment:
I think for completeness sake we should add another print statement logging which config file was read as part of this process.